### ISSUE TITLE
Rerunning Install Script after reboot should not be necessary

### ISSUE THREAD

**Issue by JPRuskin** [chocolatey/choco#3327]
### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my request.

### Is Your Feature Request Related To A Problem? Please describe.

The user-experience when installing Chocolatey on a system that lacks dotnet 4.8 is mildly annoying, and our most used flow requires:
- Running the install script
- Dotnet 4.8 being installed
- Being prompted to reboot
- Rebooting
- Running the install script again

This is less than ideal, introduces complexity, and could be improved.

### Describe The Solution. Why is it needed?

I would suggest,  given that the rest of the install seems to work fine without the reboot, that we proceed with the rest of the installation after the dotnet installation.

This would result in a system that, though needing a reboot, will come up from the reboot with a fully functional Chocolatey install - and no further required steps.

Given that Chocolatey itself prevents users from running any command beyond -v/--version when 4.8 is not installed correctly (and has an appropriately explicit error message), this should not introduce a particularly awful level of confusion.

It shouldn't break any automation, either, as the install is fairly idempotent so anyone who already is running the script, rebooting, and running it again should just work.

### Additional Context

Example of a modified install proceeding after dotnet is installed (but no reboot):
![image](https://github.com/chocolatey/choco/assets/1975761/f7e34f5d-f517-4d1f-9f35-b75df866d125)

Example of the current error message produced when there has not been a reboot:
![image](https://github.com/chocolatey/choco/assets/1975761/ef6ac466-b8f5-48d6-990b-a05279d40f71)

**Comment by JPRuskin** [comment]
I believe when I originally spoke to people about this I suggested that we could make it non-default behaviour, behind a switch or environment variable (similar to using Windows compression instead of 7zip), but as I have encountered this issue more I have decided that this is _not_ a breaking change and is fine to be the default behaviour.

**Comment by leonmelein** [comment]
I'm encountering this problem with a client at the moment. They supply VMs based on Windows Server 2019 Datacenter edition as sandboxes to data scientists. Chocolatey is deployed via PowerShell DSC but installation fails since Chocolatey 2.x for this exact reason. Rebooting and trying again hasn't proven a reliable solution. Merging the related PR should solve our issues.

**Comment by pauby** [comment]
@leonmelein have you tried installing .NET 4.8, rebooting and then installing Chocolatey CLI?

**Comment by leonmelein** [comment]
@pauby The Chocolatey installation is performed by means of a [custom script extension](https://learn.microsoft.com/en-us/azure/virtual-machines/extensions/custom-script-windows) in Azure, performed after deployment of the VM. The extension pull the latest version of Chocolatey from an internal repository and runs the install script. 

Custom script extensions cannot handle reboots in Azure, [as per Microsoft](https://learn.microsoft.com/en-us/azure/virtual-machines/extensions/custom-script-windows#:~:text=Don%27t%20put%20restarts%20inside%20the%20script.%20This%20action%20causes%20problems%20with%20other%20extensions%20that%20are%20being%20installed%2C%20and%20the%20extension%20doesn%27t%20continue%20after%20the%20restart.):

> Don't put restarts inside the script. This action causes problems with other extensions that are being installed, and the extension doesn't continue after the restart.

Given the circumstances at this client, I cannot change the way the machines are provisioned, e.g. providing a custom image or using a more recent Windows Server image.

Hope that gives a little more clarity. When I perform the installation manually, everything works as advertised.

**Comment by pauby** [comment]
@leonmelein that helps.

If you install. NET 4.8 you need to reboot before you can install Chocolatey CLI. If you install Chocolatey CLI then it will install .NET 4.8 and require you to reboot before you can use it.

In both situations you need to reboot, which you can't do.

Can you explain how this PR will fix your situation? On the surface it sounds like the requirement for .NET 4.8, however it's done, is going to have the same result?

**Comment by leonmelein** [comment]
Yes, eventually a reboot will be necessary. If the Chocolatey installation itself isn't held up by the pending reboot, I could  make sure the reboot happens at the very end of the VM deployment. That would be by using the [AzureCLI@2 ](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/azure-cli-v2?view=azure-pipelines) task and the `az vm restart` command.

At the moment, this isn't an option since there are several custom script extensions being executed sequentially directly after the deployment of the VM. This is managed completely by Azure and I cannot mix in an Azure CLI command while these extensions are running.

**Comment by pauby** [comment]
This maybe a dumb question.

Could you not install .NET 4.8 as part of the VM deployment, then reboot? Then as part of the provisioning (which I assume happens afterwards) install Chocolatey CLI, software, etc.?

**Comment by leonmelein** [comment]
That's not a dumb question. The issue with these custom script extensions is that they're executed as part of the deployment, not as separate actions. In theory we might be able to refactor this process and use a different approach, but given we're using scripts provided by another internal team that would require significant effort and coordination.

**Comment by pauby** [comment]
Thanks for that.

Just to set expectations, _any_ change to the Chocolatey CLI installation script goes through extensive testing with all products that use it. So this isn't something that is likely to be implemented soon.
